lib/CMakeLists.txt

```diff
@@ -1 +1,17 @@
-add_subdirectory(gc-dialects)
\ No newline at end of file
+add_subdirectory(gc-dialects)
+
+file(GLOB SOURCES "*.*pp")
+add_library(graph_compiler SHARED
+        ${SOURCES}
+)
+add_library(graph_compiler_static STATIC
+        ${SOURCES}
+)
```

Comment:
What are we building the two libraries for?

Reply:
The static is used by tests, because in the shared lib only the public API functions are exposed.